src/platform/networking/node/chatStream.ts

```diff
@@ -12,12 +12,81 @@ import { APIJsonData, CAPIChatMessage, ChatCompletion, rawMessageToCAPI } from '
 import { FinishedCompletion, convertToAPIJsonData } from './stream';
 
 // TODO @lramos15 - Find a better file for this, since this file is for the chat stream and should not be telemetry related
-export function sendEngineMessagesTelemetry(telemetryService: ITelemetryService, messages: CAPIChatMessage[], telemetryData: TelemetryData) {
+export function sendEngineMessagesLengthTelemetry(telemetryService: ITelemetryService, messages: CAPIChatMessage[], telemetryData: TelemetryData, logService?: ILogService) {
+	// Determine if this is input or output based on message characteristics
+	const isOutput = messages.length === 1 && messages[0].role === 'assistant';
```

Comment:
Instead of trying to guess this from the input would it be easier to just pass a flag as a parameter or something?